src/linked_list.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+"""Create a singly linked list."""
+from __future__ import unicode_literals
+
+
+class Node(object):
+    """Create a node to store data."""
+
+    def __init__(self, value, pointer=None):
+        """Create an instance of a Node."""
+        self.value = value
+        self.pointer = pointer
+
+
+class LinkedList(object):
+    """Create a link list class to store strings of data."""
+
+    def __init__(self, value_list=None):
+        """Create an instance of a singly-linked list."""
+        self.length = 0
+        self.header = None
+        if hasattr(value_list, '__iter__'):
+            for value in value_list:
+                self.push(value)
+        elif value_list:
+            self.push(value_list)
+
+    def __len__(self):
+        """Return the length of the linked list for the built in len."""
+        return self.length
+
+    def push(self, value):
+        """Add a new node to the head of the linked list."""
+        temp_node = self.header
+        new_node = Node(value, temp_node)
```

Comment:
Any reason to use this `temp_node` variable?  I'd prefer you just pass in `self.head`.